tests/components/incomfort/test_config_flow.py

```diff
@@ -0,0 +1,188 @@
+
+from aiohttp import ClientResponseError
+import pytest
+
+from homeassistant import config_entries
+from homeassistant.components.incomfort import DOMAIN
+from homeassistant.const import CONF_PASSWORD
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResultType
+
+from tests.common import MockConfigEntry
+
+MOCK_CONFIG = {
+    "host": "192.168.1.12",
+    "username": "admin",
+    "password": "verysecret",
+}
+
+
+async def test_form(
+    hass: HomeAssistant, mock_setup_entry: AsyncMock, mock_incomfort: MagicMock
+) -> None:
+    """Test we get the form."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    assert result["type"] is FlowResultType.FORM
+    assert result["errors"] == {}
+
+    result2 = await hass.config_entries.flow.async_configure(
+        result["flow_id"], MOCK_CONFIG
+    )
+    await hass.async_block_till_done()
+
+    assert result2["type"] is FlowResultType.CREATE_ENTRY
+    assert result2["title"] == "Intergas InComfort/Intouch Lan2RF gateway"
+    assert result2["data"] == MOCK_CONFIG
+    assert len(mock_setup_entry.mock_calls) == 1
+
+
+async def test_import(
+    hass: HomeAssistant, mock_setup_entry: AsyncMock, mock_incomfort: MagicMock
+) -> None:
+    """Test we get the form."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_IMPORT}, data=MOCK_CONFIG
+    )
+    await hass.async_block_till_done()
+    assert result["type"] is FlowResultType.CREATE_ENTRY
+    assert result["title"] == "Intergas InComfort/Intouch Lan2RF gateway"
+    assert result["data"] == MOCK_CONFIG
+    assert len(mock_setup_entry.mock_calls) == 1
+
+
+@pytest.mark.parametrize(
+    ("status_code", "abort_reason"),
+    [
+        (401, "auth_error"),
+        (404, "not_found"),
+        (500, "unknown_error"),
+    ],
+)
+async def test_import_client_response_fails(
+    hass: HomeAssistant,
+    mock_setup_entry: AsyncMock,
+    mock_incomfort: MagicMock,
+    status_code: str,
+    abort_reason: str,
+) -> None:
+    """Test we get the form."""
+    # pylint: disable-next=import-outside-toplevel
+    from incomfortclient import IncomfortError
+
+    mock_incomfort().heaters.side_effect = IncomfortError(
+        ClientResponseError(None, None, status=status_code)
+    )
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_IMPORT}, data=MOCK_CONFIG
+    )
+    await hass.async_block_till_done()
+    assert result["type"] is FlowResultType.ABORT
+    assert result["reason"] == abort_reason
+    assert len(mock_setup_entry.mock_calls) == 0
+
+
+@pytest.mark.parametrize(
+    ("exc", "abort_reason"),
+    [
+        (TimeoutError, "timeout_error"),
+        (ValueError("some value error"), "unknown_error"),
+    ],
+)
+async def test_import_data_fails(
```

Comment:
What's the difference between these 2 tests? Maybe you can just merge them and just add the `IncomfortError(
        ClientResponseError(None, None, status=status_code)
    )` to the list of exceptions to test

Reply:
merged